### PR TITLE
feat(config): add config

### PR DESCRIPTION
**Pull request by oysteinsigholt**

### CODE REVIEW

File: lib/clean.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const config = require("../config/config");
```
Comment (pkkummermo) [inline lib/clean.js]: This can be replaced by using [del-cli](https://github.com/sindresorhus/del-cli) without the `-f `-flag

File: package.json [hunk 2]
```diff
@@ -4,10 +4,10 @@
   "description": "Scaffold repo which documentations can fork from",
   "main": "index.js",
   "scripts": {
-    "clean": "rm -rf build/",
+    "clean": "node lib/clean.js",
```
Comment (pkkummermo) [inline package.json]: See comment in `clean.js`

File: lib/build-style.js [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
     outputStyle: "compressed"
 }, (err, result) => {
     if (err) {
         throw err;
     } else {
-        if (!fs.existsSync("./build/style/")) {
-            fs.mkdirSync("./build/style/");
-        }
-        fs.writeFile("./build/style/style.css", result.css, (error) => {
+        const targetDir = path.dirname(path.join(__dirname, "..", config.targetDir, configStyle.targetFile));
```
Comment (pkkummermo) [inline lib/build-style.js]: The package [mkdir-recursive](https://github.com/hex7c0/mkdir-recursive) does the same thing with the added bonus of tests.

File: lib/clean.js (hunk 1 above)
Comment (oysteinsigholt) [inline lib/clean.js]: How would the build directory be passed to del-cli when it is a configurable variable and not always `build`?

Comment (pkkummermo) [inline lib/clean.js]: The build dir can be static as of now, and we can improve the build tasks by adding [nps](https://github.com/kentcdodds/nps) later on which allows for more dynamic handling.

**Pull request merged**